tests/foreman/api/test_http_proxy.py

```diff
@@ -347,3 +349,62 @@ def _finalize():
     assert response.get('errors') is None
     assert repo.read().last_sync is not None
     assert repo.read().content_counts['rpm'] >= 1
+
+
+def test_refresh_updates_remotes_proxy(module_target_sat, module_org, module_product):
+    """Ensure that repo refresh updates the http-proxy of pulp remote.
+
+    :id: ab0f3734-40bd-4524-b8b9-0ab857ca3c3f
+
+    :setup:
+        1. Disable pulp CLI safe mode.
+        2. Create a product, repo and HTTP proxy.
+
+    :steps:
+        1. Get repository remote's href.
+        2. Ensure proxy_url matches the one from setup.
+        3. Set fake value to proxy_url, ensure it's written.
+        4. Refresh repos via rake and ensure the original proxy_url has been restored.
+
+    :expectedresults:
+        1. Repo refresh restores remote's original proxy_url value.
+
+    :verifies: SAT-26741
+
+    :customerscenario: true
+    """
+    sat = module_target_sat
+    sat.set_pulp_cli_safemode(False)
+
+    http_proxy = sat.api.HTTPProxy(
+        name=gen_string('alpha', 15),
+        url=settings.http_proxy.un_auth_proxy_url,
+        organization=[module_org],
+    ).create()
+
+    repo = sat.api.Repository(
+        product=module_product,
+        http_proxy_policy='use_selected_http_proxy',
+        http_proxy_id=http_proxy.id,
+    ).create()
+    sat.run_repos_refresh()
```

Comment:
Is this step necessary? My assumption is that, when setting the proxy at repo creation time, the proxy should be used by the repo without needing to explicitly run the `refresh_repos` rake task, but maybe I'm off base here.

Reply:
I'm not exactly sure why, but it seems it is necessary to set the initial pulp values. But I get your point, maybe we could dig into it deeper next week.